xsl-test/assertions.xsl

```diff
@@ -15,21 +15,26 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 -->
-<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns="http://www.w3.org/1999/xhtml" version="2.0">
+<xsl:stylesheet
+  xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+  xmlns="http://www.w3.org/1999/xhtml"
+  xmlns:xs="http://www.w3.org/1999/xhtml"
+  version="2.0"
+  exclude-result-prefixes="xs">
   <xsl:template name="assert-that">
     <xsl:param name="ignore" select="'false'"/>
     <xsl:param name="message"/>
     <xsl:param name="expected"/>
     <xsl:param name="actual"/>
     <xsl:if test="$ignore = 'false'">
-      <xsl:if test="$expected != $actual">
+      <xsl:if test="not(deep-equal($expected, $actual))">
         <xsl:message terminate="yes">
           <xsl:text>FAILURE: </xsl:text>
           <xsl:value-of select="$message"/>
           <xsl:text> (actual "</xsl:text>
-          <xsl:value-of select="$actual_xml"/>
+          <xsl:copy-of select="$actual"/>
           <xsl:text>" is not equal to expected "</xsl:text>
-          <xsl:value-of select="$expected_xml"/>
+          <xsl:copy-of select="$expected"/>
           <xsl:text>")</xsl:text>
         </xsl:message>
       </xsl:if>
```

Comment:
@paulodamaso I don't think the changes to this file belong in this PR, or?

Reply:
@amihaiemil Same as above.